openhtf/io/output/mfg_inspector.py

```diff
@@ -430,18 +447,30 @@ def UploadTestRun(testrun, destination, credentials=None):
         raise UploadFailedError(resp, content)
 
   def __call__(self, test_record):  # pylint: disable=invalid-name
-    credentials = oauth2client.client.SignedJwtAssertionCredentials(
-        service_account_name=self.user,
-        private_key=self.keydata,
-        scope=self.SCOPE_CODE_URI,
-        user_agent='OpenHTF Guzzle Upload Client',
-        token_uri=self.token_uri)
-    credentials.set_store(self._MemStorage())
-
     testrun = _TestRunFromTestRecord(test_record)
+
     try:
+      credentials = oauth2client.client.SignedJwtAssertionCredentials(
+          service_account_name=self.user,
+          private_key=self.keydata,
+          scope=self.SCOPE_CODE_URI,
+          user_agent='OpenHTF Guzzle Upload Client',
+          token_uri=self.token_uri)
+      credentials.set_store(self._MemStorage())
       self.UploadTestRun(testrun, self.DESTINATION_URL, credentials)
-    except UploadFailedError:
+    except:
       # For now, just log the exception.  Once output is a bit more robust,
       # we can propagate this up and handle it accordingly.
+      sys.stdout.write('%s\n' % self._error_message)
+      if self._fallback_filename_pattern:
```

Comment:
I propose making this a separate object:

* ` UploadToMfgInspector` shouldn't catch any errors, since now it can raise an error and it'll get logged at the top level.
* The new object (`UploadOrOutput` for instance) should create two internal objects, `UploadToMfgInspector` and `OutputToJSON`
* In the new object's `__call__`, it should try-except around calling the `Upload...` object and in the except call the `OutputToJSON` object